### PR TITLE
Update DllMap doc

### PR DESCRIPTION
**Pull request by swaroop-sridhar**
Update DllMap design doc:
* To note the NativeLibrary APIs that can be used to implement DllMap
* To point to a sample implementation of DllMap.

### CODE REVIEW

File: tests/src/Interop/DllMap/DllMap.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+using System;
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/DllMap/DllMap.cs]: License banner

**Comment by jkotas** [comment]
Samples should be in the https://github.com/dotnet/samples repo, and linked from the docs. Please submit the PR to the docs repo instead.

**Comment by jkotas** [comment]
(You can keep the design doc update if you would like.)

**Comment by swaroop-sridhar** [comment]
@Jkotas, I added the test in CoreCLR repo, because the sample builds a native library, and the Samples repo is setup with [instructions](https://github.com/dotnet/samples#building-a-snippet-or-sample) to build managed apps. 

OK, I'll move the code to samples to samples repo, with additional instructions to build the native component.

CC: @jkoritzinsky

**Comment by swaroop-sridhar** [comment]
I submitted the code to samples repo in this PR: https://github.com/dotnet/samples/pull/668 
Once that's checked in, I'll update the links in the document. Thanks.

**Pull request merged**